### PR TITLE
[JAV-566] report type/version when register to SC

### PR DESCRIPTION
**Pull request by weichao666**
1. Sdk report their framework type and version when register to SC
2. Each service with specific version has a certain framework+version
3. Present the type/version information in console in Service+Version level
4. In gov service, send the different command to CC according to the type/version when doing management

### CODE REVIEW

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/base/ServiceCombConstants.java [hunk 1]
```diff
@@ -49,4 +49,12 @@
...
 
   String CONFIG_KEY_SPLITER = "_";
+
+  String CONFIG_MICROSERVICE_DEVELOPMENT_FRAMEWORK_KEY = "framework.name";
+
+  String DEFAULT_MICROSERVICE_DEVELOPMENT_FRAMEWORK = "UNKNOWN";
+
+  String CONFIG_MICROSERVCIE_DEVELOPMENT_FRAMEWORK_VERSION_KEY = "framework.version";
+
+  String CONFIG_MICROSERVICE_REGISTER_WAY_KEY = "registreBy";
```
Comment (WillemJiang) [inline foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/base/ServiceCombConstants.java]: It should be registeredBy

File: service-registry/src/main/java/io/servicecomb/serviceregistry/api/registry/Microservice.java [hunk 2]
```diff
@@ -33,6 +33,10 @@
 public class Microservice {
   private String serviceId;
 
+  private Map<String, String> framework = new HashMap<>();
+  
+  private String registerBy;
```
Comment (WillemJiang) [inline service-registry/src/main/java/io/servicecomb/serviceregistry/api/registry/Microservice.java]: same typo here.

File: service-registry/src/main/java/io/servicecomb/serviceregistry/api/registry/Microservice.java [hunk 3]
```diff
@@ -78,6 +82,22 @@ public void setServiceId(String serviceId) {
...
 
+  public Map<String, String> getFramework() {
+    return framework;
+  }
+
+  public void setFramework(Map<String, String> framework) {
+    this.framework = framework;
+  }
+
+  public String getRegisterBy() {
```
Comment (WillemJiang) [inline service-registry/src/main/java/io/servicecomb/serviceregistry/api/registry/Microservice.java]: typo

File: foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/base/ServiceCombConstants.java (hunk 1 above)
Comment (weichao666) [reply foundations/foundation-common/src/main/java/io/servicecomb/foundation/common/base/ServiceCombConstants.java]: Sorry, the spelling is wrong here,it has been changed to registerBy,it's sure better to use registeredBy, but this field is provided by SC, i have to align with them...

**Pull request closed without merging**